src/driver/compile.ml

```diff
@@ -67,20 +69,62 @@ let init_stats (pkgs : Packages.t Util.StringMap.t) =
 
 open Eio.Std
 
-let compile output_dir all =
+type partial =
+  (string * compiled) list * (string * Packages.modulety) Util.StringMap.t
+
+let unmarshal filename =
+  let ic = open_in_bin (Fpath.to_string filename) in
+  let (v : partial) = Marshal.from_channel ic in
+  close_in ic;
+  v
+
+let marshal (v : partial) filename =
+  let p = Fpath.parent filename in
+  Util.mkdir_p p;
+  let oc = open_out_bin (Fpath.to_string filename) in
+  Marshal.to_channel oc v [];
+  close_out oc
+
+let find_partials odoc_dir =
+  let tbl = Hashtbl.create 1000 in
+  let hashes_result =
+    Bos.OS.Dir.fold_contents ~dotfiles:false
```

Comment:
This could just iterate on directories and check whether they contain a `index.m` file with `` ~elements:`Dirs ``.